frontend/src/app/organization/organization-create/organization-create.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AuthService } from "../../auth.service";
+
+import { Organization } from '../organization.model';
+import { OrganizationService } from '../organization.service';
+
+@Component({
+  selector: 'volontulo-organization-create',
+  templateUrl: './organization-create.component.html',
+  styleUrls: ['./organization-create.component.scss']
+})
+export class OrganizationCreateComponent implements OnInit {
+  createForm: FormGroup;
+  id: number;
+  inEditMode = false;
+  message = '';
+
+  constructor(
+    private authService: AuthService,
+    private activatedRoute: ActivatedRoute,
+    private fb: FormBuilder,
+    private organizationService: OrganizationService,
+    private router: Router,
+    ) {}
+
+  ngOnInit() {
+    this.createForm = this.fb.group({
+       'name': this.fb.control(null, Validators.required),
+       'address': this.fb.control(null, Validators.required),
+       'description': this.fb.control(null, Validators.required),
+      });
+    this.activatedRoute.params.subscribe(
+      params => {
+        if (params.organizationId) {
+          this.inEditMode = true;
+          this.id = params.organizationId;
+          this.organizationService.getOrganization(params.organizationId);
+          this.organizationService.organization$
+            .subscribe((organization: Organization) => {
+            this.createForm.patchValue(organization);
+          });
+        }
+    });
+    this.organizationService.createStatus$
+      .subscribe(
+        (response) => {
+          if (response.status === 'success') {
+            this.authService.getCurrentUser();
```

Comment:
This line looks really suspicious to me. Could you explain what was the problem you encountered that you decided to add this line? My guess is that you wanted the user to be logged in - is that the case here?

Reply:
Adding an organization changes user's profile and I wanted somehow fetch the current user from the server once again because the list of organizations for user changed and it determines the way how things in organization's detail view are displayed (this component is loaded after adding new organization and without "user's refreshing" the logic related to `isUserOrgMember` doesn't work correctly).